### ISSUE TITLE
Project definition invalid with pipenv on Python 3.8.9

### ISSUE THREAD

**Issue by fkrauthan** [yaml/pyyaml#596]
It seems like that when trying to install PyYAML on a pipenv `2021.11.23` with Python `3.8.9` it throws the following error:

```
pipenv.patched.notpip._internal.exceptions.InstallationError: Disabling PEP 517 processing is invalid: project specifies a build backend of setuptools.build_meta in pyproject.toml
During handling of the above exception, another exception occurred:
```

while processing the pyyaml dependency.

**Comment by nitzmahone** [comment]
This is most definitely not a pyyaml bug- take a look at https://stackoverflow.com/questions/65658570/pipenv-install-fails-on-cryptography-package-disabling-pep-517-processing-is-i for some potential solutions. The most likely case (without knowing your environment details) is that for some reason, there isn't a wheel available for your platform, and your environment is missing some deps to build pyyaml. When a PEP517 build fails, `pipenv` apparently silently tries to build again by disabling PEP517, which is not allowed by pip for projects that explicitly declare a PEP517 build config (nearly all major Python projects, these days).

Closing as not a bug.

**Comment by fkrauthan** [comment]
Hmm so what exactly is missing as the readme does not tell me anything I have to install? Because other python packages don't have that issue (including other yaml libraries).

**Comment by nitzmahone** [comment]
pipenv is ... special- it hacks around so many things, the problem could be literally anything- you'd have to turn on some more verbose logging of what pipenv's doing to know. Installing pyyaml under a proper PEP517 build environment (eg pip) will do its best to install all the necessary bits to get you something that functions, but if pipenv is mucking around and disabling that or modifying the default package index settings, all bets are off.

**Comment by fkrauthan** [comment]
Well if I understand the error correctly (and based on stuff I found searching for the error). It seem to be complaining about the `pyproject.toml` file provided by pyyaml. As it contains `build-backend = "setuptools.build_meta"` which does not seem to be valid/supported anymore? But to be honest I am not very familiar with Python. Just thought I raise it with you in case you have any ideas.

**Comment by kitterma** [comment]
The error message from pipenv is clearly wrong.  `build-backend = "setuptools.build_meta"` is exactly the correct specification for a PEP 517 based build using setuptools.

**Comment by fkrauthan** [comment]
Interesting. Yeah it's strange. It seem to happen when it tries to lock dependencies. If I skip the lock file it does install and work properly.

**Comment by nitzmahone** [comment]
The PEP517 error is a red herring- IIUC pipenv tries first to build with PEP517 enabled, then tries again by explicitly disabling it. I believe you're seeing an error patched through from pip from the second try after it gives up, where the information on what's really happening is in the output from the first try.

**Comment by m-andre** [comment]
FWIW: I am using macOS on Apple M1 chip (arm64). It really looks like a problem of missing wheel for Python 3.8 - see [https://pypi.org/simple/pyyaml/](https://pypi.org/simple/pyyaml/) - there is no wheel for cp38 and arm64. I have no problem to lock it with Python 3.9.

**Comment by nitzmahone** [comment]
We can't ship a wheel for Python 3.8- Apple's Python 3.8 won't build something that functions, and universal2/arm64 python.org builds (which is what we use for building wheels) are only available on Python 3.9+.
